core/templates/components/entity-creation-services/topic-creation.service.ts

```diff
@@ -43,22 +59,63 @@ angular.module('oppia').factory('TopicCreationService', [
             'new-topic-name-editor.template.html'),
           backdrop: true,
           controller: [
-            '$scope', '$uibModalInstance',
-            function($scope, $uibModalInstance) {
-              $scope.topicName = '';
-              $scope.abbreviatedTopicName = '';
+            '$scope', '$uibModalInstance', 'TopicEditorStateService',
+            function($scope, $uibModalInstance, TopicEditorStateService) {
+              $scope.idHasLoaded = false;
+              $scope.pageHasLoaded = false;
+              $scope.topic = TopicEditorStateService.getTopic();
+              $scope.topicRights = TopicEditorStateService.getTopicRights();
+              $scope.topicNameEditorIsShown = false;
+              $scope.editableName = $scope.topic.getName();
+              $scope.editableAbbreviatedName = (
+                $scope.topic.getAbbreviatedName());
+              $scope.editableDescription = $scope.topic.getDescription();
+              $scope.allowedBgColors = (
+                topicConstants2.ALLOWED_THUMBNAIL_BG_COLORS.topic);
+              $scope.categories = TOPIC_CATEGORIES;
+              $scope.editableDescriptionIsEmpty = (
+                $scope.editableDescription === '');
+              $scope.topicDescriptionChanged = false;
+              $scope.topicObject = {
+                name: '',
+                category: '',
+                description: '',
+              };
+              $scope.updateTopicThumbnailFilename = function(
+                  newThumbnailFilename) {
+                if (newThumbnailFilename ===
+                    $scope.topic.getThumbnailFilename()) {
+                  return;
+                }
+                TopicUpdateService.setTopicThumbnailFilename(
+                  $scope.topic, newThumbnailFilename);
+              };
+              $scope.updateTopicThumbnailBgColor = function(
+                  newThumbnailBgColor) {
+                if (newThumbnailBgColor ===
+                    $scope.topic.getThumbnailBgColor()) {
+                  return;
+                }
+                TopicUpdateService.setTopicThumbnailBgColor(
+                  $scope.topic, newThumbnailBgColor);
+              };
               $scope.MAX_CHARS_IN_TOPIC_NAME = MAX_CHARS_IN_TOPIC_NAME;
+              $scope.MAX_CHARS_IN_TOPIC_DESCRIPTION = (
+                MAX_CHARS_IN_TOPIC_DESCRIPTION);
               // No need for a length check below since the topic name input
               // field in the HTML file has the maxlength attribute which
               // disallows the user from entering more than the valid length.
-              $scope.isTopicNameValid = function() {
-                return $scope.topicName !== '';
+              $scope.isTopicValid = function() {
+                for (let key in $scope.topicObject) {
```

Comment:
This should be part of a domain object function for the topicObject (which should be renamed if its fields are not the same as that of a Topic).

Reply:
Done.